code/modules/reagents/Chemistry-Goon-Pyrotechnics.dm

```diff
@@ -141,4 +131,44 @@ proc/goonchem_vortex(var/turf/simulated/T, var/setting_type, var/range, var/pull
 						step_away(X,T)
 				else
 					for(var/i = 0, i < pull_times, i++)
-						step_towards(X,T)
\ No newline at end of file
+						step_towards(X,T)
+
+/datum/reagent/blackpowder
+	name = "Black Powder"
+	id = "blackpowder"
+	description = "Explodes. Violently."
+	reagent_state = LIQUID
+	color = "#000000"  //rgb: 96, 165, 132
+	metabolization_rate = 0.05
+
+/datum/chemical_reaction/blackpowder
+	name = "Black Powder"
+	id = "blackpowder"
+	result = "blackpowder"
+	required_reagents = list("saltpetre" = 1, "charcoal" = 1, "sulfur" = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/blackpowder_explosion
+	name = "Black Powder Kaboom"
+	id = "blackpowder_explosion"
+	result = null
+	required_reagents = list("blackpowder" = 1)
+	result_amount = 1
+	required_temp = 474
+	mix_message = "<span class = 'userdanger'>Sparks start flying around the black powder!</span>"
+
+/datum/chemical_reaction/blackpowder_explosion/on_reaction(var/datum/reagents/holder, var/created_volume)
+	sleep(rand(10,30))
+	var/turf/simulated/T = get_turf(holder.my_atom)
+	var/ex_severe = round(created_volume / 10)
+	var/ex_heavy = round(created_volume / 8)
+	var/ex_light = round(created_volume / 6)
+	var/ex_flash = round(created_volume / 4)
+	if(ex_severe > 3)
+		ex_severe = 3
+	if(ex_heavy > 7)
+		ex_heavy = 7
+	if(ex_light > 14)
+		ex_light = 14
+	explosion(T,ex_severe,ex_heavy,ex_light,ex_flash, 1, 1)
+	return
```

Comment:
i'll increase the delay but i'm not gonna do the divisors, anything above what I had now resulted in 200u being a pitifully small explosion